here-naksha-app-service/src/main/resources/swagger/openapi.yaml

```diff
@@ -476,25 +476,97 @@ paths:
           $ref: '#/components/responses/ErrorResponse504'
         '513':
           $ref: '#/components/responses/ErrorResponse513'
+    put:
+      tags:
+        - Write Features
+      summary: Update features in the space
```

Comment:
I have a question here: the current description states that this is an atomic operation, so either all features will be upserted, or non at all. Is this still the intended design for these APIs, even with the new class ´ResultCursor´ now for supporting partial modification?

Reply:
Very good question. We need to validate that this is how API is also behaving. Perhaps a test case, where one feature is successful and other one fails due to UUID conflict. Ensure at the end, even the success one is rollback'ed.
Actually, this is what needs to be handled at Hub side. When Hub calls lib-psql `execute(WriteRequest)` and when Hub finds from the `Result` that the operation has failed (even if partially), then Hub should explicitly rollback entire transaction. But we may not be able to validate this with mock, as mock doesn't have rollback capability, and it will not be that worth putting complex efforts building the rollback capability for a short duration and that too for one or two scenarios. But, we should certainly revalidate this when psql alignment is in place. Let's add this in our backlog.

Reply:
thanks for clarifying, I added this into the description for MCPODS-6535